pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,33 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional OpenShift components are specified in the
+// install-config to be included in the installation.
+func validateCapabilities(c *configv1.ClusterVersionCapabilitiesSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; !found {
+		allErrs = append(allErrs, field.Invalid(fldPath, c.BaselineCapabilitySet, "unknown baseline capability set specified"))
+	}
+
+	allAvailableCapabilities := map[string]bool{}
+	allCaps := []string{}
+	// Create a map and slice of *all* available capabilities across all capability sets
+	for k, _ := range configv1.ClusterVersionCapabilitySets {
+		for _, capability := range configv1.ClusterVersionCapabilitySets[k] {
```

Comment:
nit: you should be able to use:

```go
for _, capabilities := range configv1.ClusterVersionCapabilitySets {
    for _, capability := range capabilities {
```

to avoid having to touch the keys or lookup from the map twice.

Reply:
Fixed in a10be333d